### PR TITLE
[ready?] Disallows pacifists from using the HONK Punching Glove, but adds a non-lethal toggle.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Now that the HONK can break people's ribs and bones, I figured I should do something about this.

Previously, the HONK Punching Glove equipment wasn't marked as harmful, meaning pacifists could use it do deal 35 damage + stunlock (+break ribs and bones now). I removed that, because it's an exploit to get around pacifism to kill people. 

Now, the punching glove has a toggle between lethal and non-lethal punching, and the distance it throws you scales on the punching damage. Also included some vars for admin editing, for fun. 

Yes, i'm aware the you can still 'damage' people if you hit people into solid walls. It's much less damage though, so i'm not too concerned.
## Why It's Good For The Game

Removing an exploit to get around pacifism is good.
I thought people would complain if I made it harmful and left it at that, so I decided a toggle would be the best of both worlds - this way, pacifist clowns can still push people around and prank them, but they won't break ribs. 

## Changelog
:cl: Melbert
add: Allows the HONK Oingo-Boingo Punch-Face to toggle between lethal and non-lethal modes. 
tweak: The HONK Oingo-Boingo Punch-Face's throw distance now scales on how much damage it deals. By default, it's the same distance as before.
fix: Pacifists can no longer use the HONK Oingo-Boingo Punch-Face to damage people.
admin: Admins can now edit the damage of the HONK Oingo-Boingo Punch-Face. Honk.
/:cl:

### CODE REVIEW

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 1]
```diff
@@ -463,14 +463,35 @@
...
+		return "[..()] &nbsp; <a href='?src=[REF(src)];lethalPunch=1'>[harmful?"Punch":"Pat"] mode</a>"
+	else
+		return
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/punching_glove/Topic(href, href_list)
+	..()
+	if(href_list["lethalPunch"])
+		harmful = !harmful
+		if(harmful)
+			chassis?.occupant_message("<span class='warning'>Lethal Fisting Enabled.</span>")
```
Comment (Iamgoofball) [inline code/game/mecha/equipment/weapons/weapons.dm:491]: while funny, our players cannot be trusted with fisting
https://github.com/tgstation/tgstation/pull/40941

Comment (MrMelbert) [reply code/game/mecha/equipment/weapons/weapons.dm:491]: what a clown does in the privacy of his own HONK is none of my business 😳 (it only shows up to the mech pilot so it's probably fine)

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 2]
```diff
@@ -463,14 +463,36 @@
 	fire_sound = 'sound/items/bikehorn.ogg'
 	projectiles = 10
 	projectile_energy_cost = 500
+	harmful = TRUE
 	diags_first = TRUE
+	var/punch_damage = 35 ///damage done by the glove on contact
+	var/can_toggle_lethal = TRUE ///TRUE - can toggled between lethal and non lethal || FALSE - cannot toggle
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/mecha/equipment/weapons/weapons.dm:469]: You have generated new variables/procs/defines without proper documentation for the autodoc to work.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

**Comment by MrMelbert** [comment]
this is probably good to go maybe, the doc is fixed

**Pull request merged**